app/Http/Controllers/EmployeeController.php

```diff
@@ -2,82 +2,154 @@
+                              ->orWhere('address' ,'like' , "%".$search."%")
+                              ->orWhere('dob' ,'like' , "%".$search."%")
+                              ->orWhere('phone' ,'like' , "%".$search."%")
+                              ->orWhereHas('bankAccount',
+                              function ($q) use ($search) {
+                                  $q->where('account_no', 'like', "%".$search."%")->select('branch'); })
+                              ->orWhereHas('company', function ($q) use ($search) {
+                                  $q->where('name', 'like', "%".$search."%"); })
+                              ->orWhereHas('company', function ($q) use ($search) {
+                                  $q->where('branch', 'like', "%".$search."%"); });
+
+          $filteredEmployees = $search ? $employees->count() : $totalEmployees;
+          $employees = $employees->skip($start)
+                                  ->take($length)
+                                  ->get();
 
-        //SERVER SIDE RENDERING HANDLE FOR data table
-        $search = $request->query('search')['value'];
-        $draw = $request->query('draw', 1);
-        $start = $request->query('start', 0);
-        $length = $request->query('length', 10);
-        $totalEmployees =   Employee::count();
-
-        $employees = Employee::with('bankAccount' , 'company')->where('name' , 'like' , "%".$search."%")
-                            ->orWhere('position' ,'like' , "%".$search."%")
-                            ->orWhere('email', 'like' , "%".$search."%")
-                            ->orWhere('address' ,'like' , "%".$search."%")
-                            ->orWhere('dob' ,'like' , "%".$search."%")
-                            ->orWhere('phone' ,'like' , "%".$search."%")
-                            ->orWhereHas('bankAccount',
-                            function ($q) use ($search) {
-                                $q->where('account_no', 'like', "%".$search."%")->select('branch'); })
-                            ->orWhereHas('company', function ($q) use ($search) {
-                                $q->where('name', 'like', "%".$search."%"); });
-
-        $filteredEmployees = $search ? $employees->count() : $totalEmployees;
-        $employees = $employees->skip($start)
-                                ->take($length)
-                                ->get();
-
-        $response = [
-            'draw' => intval($draw),
-            'recordsTotal' => intval($totalEmployees),
-            'recordsFiltered' => $filteredEmployees,
-            'data' => $employees
-        ];
-
-        return Response::json($response);
+          $response = [
+              'draw' => intval($draw),
+              'recordsTotal' => intval($totalEmployees),
+              'recordsFiltered' => $filteredEmployees,
+              'data' => $employees
+          ];
+          return Response::json($response);
     }
 
+
     public function store(Request $request){
 
```

Comment:
Please use one try catch block to identify the error

Reply:
okay sure

Reply:
it wasnt possible to validate and put both models into one try catch block because if one model fails it will throw an exception and stop executing below that and both were same exception (validationExecption) therefore i couldnt use multiple catch block to catch different exceptions too...therefore i implemented the validation in a new method by creating a manual validator which solved the issue of code redundancy.....Please review this approach and give feedbacks @Raza9798  #31

Reply:
we don't catch error using if conditions. if the validation failed. it will automatically pass to catch block where you need to return the HTTP request. use one block of try catch for entire method. @axnish27

Reply:
Yes got it wrapped into one try catch please check... @Raza9798